tracks/ruby/exercises/high-scores/mentoring.md

```diff
@@ -34,28 +34,27 @@ end
 
 ```
 ### Reasonable variants
- - Using a getter method `scores`, not `attr_reader`: That's okay at this point in the track, no need to suggest `attr_reader`.
+ - Using `attr_reader :scores` instead of the getter method `scores`: Is better, but no need to point them to `attr_reader` if they write the method `scores`. 
  - Variants in composing the string in `report`, as long as it's not a complicated if/else statement (see Talking Points).
 
-### General notes
+### General 
 - The Instructions point to a beginner friendly explanation of instantiating a class: 
-http://ruby-for-beginners.rubymonstas.org/writing_classes/initializers.html
-and to the rubydocs Array. 
-- Beginners create the attribute reader, but may call the instance variable instead of the getter. 
-This link gives a nice introduction of getters and setters, and the naming convention in Ruby:
-[launchschool: accessor_methods](https://launchschool.com/books/oo_ruby/read/classes_and_objects_part1#accessormethods).
-It's recommended to link to this or your favorite explanation, instead of typing the solution for them.
+[Ruby for Beginners](http://ruby-for-beginners.rubymonstas.org/writing_classes/initializers.html)
+and to the rubydocs [Array](https://ruby-doc.org/core/Array.html). 
 - `sort.reverse` is not very performant, but at this point in the track it's more important that students explore the 
 standard libraries (and work with method chaining) than that they know how to optimize.
 
 ### Talking points
-- [`Array#max`](https://ruby-doc.org/core/Array.html#method-i-max) takes an argument; `scores.max(3)` works.
-- Most submissions retrieve the `personal_top` values with the `Array#[]` method, which is harder to read than the convenience methods that Ruby offers: `Array#take(3)` or `Array#first(3)` win in readability. And `Array#max(3)` doesn't even need the sorting-and-reverting steps. 
-- The `report` method asks for a conditionally composed string. It's meant to think about the trade-off 'duplication' vs 'readability'. Recommendation: approve any solution unless it's a supercomplicated if/else statement. 
-- Consider `String#squeeze(' ')` to avoid hard-to-spot spacing with string interpolation. 
+- `@scores` vs `scores`: This is the most important issue in this exercise. Beginners may _write_ the getter method/attr_reader, but _call_ the instance variable `@scores` still.
+This link explains:[launchschool: accessor_methods](https://launchschool.com/books/oo_ruby/read/classes_and_objects_part1#accessormethods). It's recommended to link to this or your favorite explanation, instead of typing the solution for them. 
```

Comment:
Another point: this link is a nice resource - thank you! Maybe consider changing it to [this one](https://launchschool.com/books/oo_ruby/read/classes_and_objects_part1#accessormethodsinaction) which is the part where they address `@name` vs. `name` specifically? Or maybe it to the `@scores` vs `scores` point above?

I think this `@scores` vs `scores` point is still a somewhat dividing thing within the community. We had a small discussion about this early on the Mentor Slack, but I can't find it anymore 😞 Wish there were more blog posts about this.

It clicked on a different level for me when I read POODR. It was on Chapter 2 on the "Depend on Behavior, Not Data" section. [This summary](https://github.com/serodriguez68/poodr-notes#depend-on-behavior-methods-not-data) shows some context. I do find some of the concepts she explores to be advanced. It's interesting to see how a super simple exercise can lead to very deep conversations about OO.

Reply:
> Maybe consider changing it to this one which is the part where they address @name vs. name specifically?
I see what you mean, and I had to think hard why I didn't do that in the first place. The reason is that the article starts where the link from the instructions stops. And I believe they need the full explanation of how to come from the initializer to calling the getter/attr_reader. Instead of just the solution of how it should be done. (Because we only need to address this Talking Point if they _don't_ get that.

Does this convince you? If you still feel it will help mentors if we add the anchor link, I'll add it!

Reply:
> Does this convince you? If you still feel it will help mentors if we add the anchor link, I'll add it!

I was thinking about the `@scores` vs `scores` links as refreshers for the mentors themselves. If it's for the students, sure the one you have makes sense :+1:

>  I don't think calling the getter/attr_reader if it exists, is a controversial thing; maybe explicitly adding an attr_reader/getter_method to access the instance variable is a matter of taste/discussion. ??

I'm not sure if it's only when you need to add a getter or not. My main point was for us to add more supporting links if we had some, for mentors to read through. Anyway, I think the one you added is already very good. [Here's another interesting discussion on the topic.](https://forum.upcase.com/t/using-instance-variables-vs-attribute-accessors/1788)

Reply:
Aha, I remember now! That was a different discussion, yeah, where Joel held a passionate speech on instance variables, linked to that same post and then discovered his own comment. 
Maybe nice to add to the Mentor Research section, if/when it proves to raise questions for mentors.

Reply:
Sounds good :+1: